### DISCUSSION TITLE
Virtual ManyToOne based on another field

### DISCUSSION THREAD

**Discussion by edorgeville** [mikro-orm/mikro-orm#2604] [Q&A] [answered]
Hi,
I've been going in circle for quite a while now trying to make this work:
Is it possible to have a `ManyToOne` relationship derived from another property?
Consider the two following example entities:
```typescript
// Employee.ts
import { Entity, ManyToOne, PrimaryKey, Property } from "@mikro-orm/core"
import { ObjectId } from "@mikro-orm/mongodb"
import { Boss } from "./Boss"

@Entity()
export class Employee {
  @PrimaryKey()
  _id!: ObjectId

  @Property()
  name: string = 'John Doe'

  @Property()
  bossName: string

  // Somehow, make this work using `bossName`
  @ManyToOne(() => Boss)
  boss!: Boss

  constructor (name?: string) {
    this.name ??= name
  }
}
```
```typescript
// Boss.ts
import { Collection, Entity, OneToMany, PrimaryKey, Property } from "@mikro-orm/core"
import { ObjectId } from "@mikro-orm/mongodb"

@Entity()
export class Boss {
  @PrimaryKey()
  _id!: ObjectId

  @Property()
  name: string = 'Jeff Bedoes'

  constructor (name?: string) {
    this.name ??= name
  }
}
```

I'd like to be able to `const employee = new Employee()`, assign the boss' name to the `employee.bossName` property, and then access `employee.boss` relationship, which would return the matching `Boss` entity.
Is this achievable? I found [this comment](https://github.com/mikro-orm/mikro-orm/discussions/1210#discussioncomment-1446476) linking to [this other comment](https://github.com/mikro-orm/mikro-orm/issues/1469#issuecomment-783188128), but at this point my brain has turned into a useless blob and I can't seem to understand how to apply it to my use case.
Let me know if a complete repro would help.

Thanks in advance!

**Comment by B4nan** [comment, marked as answer]
So you basically want a query to be issued in the background to look up such `Boss` entity based on the `bossName`, every time the value is set or read, do I get that right? Otherwise how would you want to get that entity? It does not have to be loaded in the identity map, so just checking that can't be enough.

This is not supported, not sure how could it work without being async and doing real entity query. In v5 you could achieve something like this if it would be a PK (so `bossId`), but not with a name. Better to define a regular async getter and setter (regular async methods like `getBoss()` and `setBoss(name: string)`) and maybe cache the value.
